### PR TITLE
Improve check of signatures subset in stateful validator

### PR DESCRIPTION
**Pull request by steephengeorge**
### Description of the Change
Implemented the suggested review comments.

### Benefits
Better execution complexity, simplicity and performance

### Possible Drawbacks 
There is no drawbacks

### CODE REVIEW

**Comment by l4l** [comment]
Hmm, it seems that we doesn't have a test case. Do you mind to write some?

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 1]
```diff
@@ -95,18 +95,14 @@ namespace iroha {
...
-      }
-      return std::all_of(public_keys.begin(),
+            
+	   return std::all_of(public_keys.begin(),
                          public_keys.end(),
-                         [&txPubkeys](const auto &public_key) {
-                           return txPubkeys.find(public_key.toString())
-                               != txPubkeys.end();
+                         [&signatures](const auto &public_key) {
+                           return signatures.find(public_key)
```
Comment (l4l) [inline irohad/validation/impl/stateful_validator_impl.cpp]: Signatures has has a key `types::SignatureType`, not `crypto::PublicKey`, that won't compile

**Comment by lebdron** [comment]
@steephengeorge any updates on this?

**Comment by lebdron** [comment]
Closing due to lack of activity in a month, please reopen when ready.

**Pull request closed without merging**